### ISSUE TITLE
Enhancement request: add custom UIView in place of SwipeActionButton (UIButton)

### ISSUE THREAD

**Issue by MarcoFilosi** [SwipeCellKit/SwipeCellKit#162]
It would be nice to have SwipeAction more customizable by adding a custom `UIVIew` instead of using the `SwipeActionButton` (`UIButton`) mandatory.

Please see also this issue (#160) and this pull request (#161). It was a first attempt to add a `UISwitch` along with the `SwipeActionButton`.

**Comment by MarcoFilosi** [comment]
@kurabi I agree with you to have this long term solution. By the way I think we should do a code refactoring before having this new feature in place.  At the moment code is too tied to the `SwipeActionButton` class. What do you think?

**Comment by SureshSc** [comment]
@kurabi any idea when this will be done?
